test/functional/feature_proxy.py

```diff
@@ -279,11 +284,9 @@ def networks_dict(d):
         if self.have_ipv6:
             n3 = networks_dict(self.nodes[3].getnetworkinfo())
             assert_equal(NETWORKS, n3.keys())
+            proxy = f'[{self.conf3.addr[0]}]:{self.conf3.addr[1]}'
             for net in NETWORKS:
-                if net == NET_I2P or net == NET_ONION:
-                    expected_proxy = ''
-                else:
-                    expected_proxy = f'[{self.conf3.addr[0]}]:{self.conf3.addr[1]}'
+                expected_proxy = '' if net == NET_I2P or net == NET_ONION else proxy
```

Comment:
nit: this looks unnecessary coz the expression is used just once. Also, personally, I find the perl-style harder to read: `x=1 if some_condition else 2`.

I am fine as it is, but I will be happier if commit `test: hoist proxy out of 2 network loops in feature_proxy.py` is removed :)

Reply:
This line is invoked 3 times in the loop, so it seems better to do the more complex lookups and string interpolation before entering it. The ternary is used frequently in our Python tests and I think it's more succinct here to use the ternary form for an assignment instead of if - else with two assignments.